### PR TITLE
Neumann bc

### PR DESCRIPTION
**Pull request by lxmota**
Start implementation of traction BCs for Schwarz contact.

### CODE REVIEW

File: src/nimble_boundary_condition_manager.h [hunk 1]
```diff
@@ -92,7 +92,7 @@ class BoundaryConditionManager
 
   template <typename ViewT>
   void
-  ApplyInitialConditions(const ViewT reference_coordinates, ViewT velocity, std::vector<ViewT>& offnom_velocities)
+  ApplyInitialConditions(const ViewT reference_coordinates, ViewT velocity)
```
Comment (uhetmaniuk) [inline src/nimble_boundary_condition_manager.h]: I believe the variable 'offnom_velocities' is used when the code is running a UQ simulation.

@rjones-sandia may confirm.

File: src/nimble_boundary_condition_manager.h [hunk 2]
```diff
@@ -116,32 +111,20 @@ class BoundaryConditionManager
...
-
   template <typename ViewT>
   void
   ApplyKinematicBC(
-      double              time_current,
-      double              time_previous,
-      const ViewT         reference_coordinates,
-      ViewT               displacement,
-      ViewT               velocity,
-      std::vector<ViewT>& offnom_velocities)
```
Comment (uhetmaniuk) [inline src/nimble_boundary_condition_manager.h]: Same comment about UQ

**Review by uhetmaniuk — CHANGES_REQUESTED** [review]
I agree with removing the "*_fluctuation*" variables.  They were only used by the RVE code.

I believe the "offnom_*" variables are related to UQ.

**Comment by lxmota** [comment]
Weren't we removing UQ as well? Also, those velocities were not doing anything at all.

**Comment by uhetmaniuk** [comment]
UQ has been moved in a sub-directory. 
The capability was not planned to be removed.
Reese and his collaborator are fixing it now.

**Comment by uhetmaniuk** [comment]
@lxmota The overload is here https://github.com/NimbleSM/NimbleSM/blob/4e57816e1debad17650d202302370b2fee42ef2c/src/nimble_boundary_condition_manager.h#L130

and for kinematic conditions,
https://github.com/NimbleSM/NimbleSM/blob/4e57816e1debad17650d202302370b2fee42ef2c/src/nimble_boundary_condition_manager.h#L208

**Comment by rjones-sandia** [comment]
Urich is correct - we need the off_nom_* fields & we are currently revitalizing UQ.

======================================================================
"the second law is just a fancy way of saying things are going from bad to worse" -- Eeyore
"nah, they are just getting more interesting" -- Piglet
======================================================================
Reese Jones (925) 294-4744

________________________________
From: Alejandro Mota ***@***.***>
Sent: Wednesday, July 28, 2021 7:15 PM
To: NimbleSM/NimbleSM ***@***.***>
Cc: Jones, Reese ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [NimbleSM/NimbleSM] Neumann bc (#262)


Weren't we removing UQ as well? Also, those velocities were not doing anything at all.

**Comment by lxmota** [comment]
Ok, I just reverted the commits that removed those variables.

File: src/nimble_boundary_condition_manager.h (hunk 1 above)
Comment (lxmota) [reply src/nimble_boundary_condition_manager.h]: Reverted commits that removed these variables.

**Review by uhetmaniuk — APPROVED** [review]

**Review by uhetmaniuk — APPROVED** [review]
Looks good.

**Pull request merged**